src/runtime_src/hip/core/device.h

```diff
@@ -3,6 +3,10 @@
 #ifndef xrthip_device_h
 #define xrthip_device_h
 
+#ifdef _WIN32
+#undef max
+#endif
```

Comment:
This is annoying yes.   But I think the better fix is to #define NOMINMAX before inclusion of windows.h in all units that include this header.  I have seen this in Microsoft's own code:
```
#define NOMINMAX
#include <windows.h>
```

Reply:
Hi @stsoe, when I remove this I get some different errors from MSVC internal code. So I haven't removed this. Hope that is okay